src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const value = item[key];
+
+    if (Object.prototype.hasOwnProperty.call(groupedByKey, value)) {
+      groupedByKey[value].push(item);
+    } else {
+      groupedByKey[value] = [item];
+    }
```

Comment:
you can invert your condition to just **add blank array**, and **then** push your item to get rid of `else`